packages/dds/sequence/src/intervalIndex/overlappingIntervalsIndex.ts

```diff
@@ -4,12 +4,8 @@
  */
 
 import { Client } from "@fluidframework/merge-tree";
-import {
-	IntervalType,
-	IIntervalHelpers,
-	IntervalIndex,
-	ISerializableInterval,
-} from "../intervalCollection";
+import { IntervalIndex } from "../intervalCollection";
```

Comment:
would it make sense to move the IntervalIndex interface (and anything else that deals with interval indexes) out of intervalCollection.ts and into the new intervalIndex directory?

Reply:
Yes indeed, a separate PR will be opened to handle the IntervalIndex-related API. This PR is solely focused on refactoring the interval-related interfaces for the easier review.